tests/platform_tests/cli/test_show_platform.py

```diff
@@ -30,42 +31,111 @@
+    #    hwsku based on 'hwsku' or 'sonic_hwsku' inventory variable.
+    #    platform based on 'sonic_hw_platform'  inventory variable.
+    #    asic based on 'asic_type'  inventory variable.
+    #    num_asic on 'num_asics' inventory variable
+    expected_hwsku = dut_vars['hwsku'] if 'hwsku' in dut_vars else None
+    if not expected_hwsku:
+        # Lets try 'sonic_hwsku' as well
+        expected_hwsku = dut_vars['sonic_hwsku'] if 'sonic_hwsku' in dut_vars else None
+    expected_platform = dut_vars['sonic_hw_platform'] if 'sonic_hw_platform' in dut_vars else None
+    expected_asic = dut_vars['asic_type'] if 'asic_type' in dut_vars else None
+    expected_num_asic = str(dut_vars['num_asics']) if 'num_asics' in dut_vars else None
+
+    expected_fields_values = {expected_platform, expected_hwsku, expected_asic}
+    if len(unexpected_fields) != 0:
+        expected_fields_values.add(expected_num_asic)
+
+    actual_fields_values = set(summary_dict.values())
+    diff_fields_values = expected_fields_values.difference(actual_fields_values)
+    pytest_assert((len(diff_fields_values) == 0 or (len(diff_fields_values) == 1 and diff_fields_values.pop() is None)),
+                  "Unexpected value of fields, actual=%s, expected=%s on host '%s'" % (
+                        str(actual_fields_values), str(expected_fields_values), duthost.hostname))
+
+
+def test_show_platform_syseeprom(duthosts, enum_rand_one_per_hwsku_hostname, dut_vars):
     """
     @summary: Verify output of `show platform syseeprom`
     """
-    duthost = duthosts[rand_one_dut_hostname]
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
     cmd = " ".join([CMD_SHOW_PLATFORM, "syseeprom"])
 
-    logging.info("Verifying output of '{}' ...".format(cmd))
-    syseeprom_output = duthost.command(cmd)["stdout"]
-    # TODO: Gather expected data from a platform-specific data file instead of this method
+    logging.info("Verifying output of '{}' on '{}' ...".format(cmd, duthost.hostname))
+    syseeprom_cmd = duthost.command(cmd)
+    syseeprom_output = syseeprom_cmd["stdout"]
+    syseeprom_output_lines = syseeprom_cmd["stdout_lines"]
+
+    """
+    Gather expected data from a inventory file instead if 'syseeprom_info' is defined in the inventory
+    # Sample inventory with syseeprom:
+    
+    str- msn2700-01:
+        ansible_host: 10.251.0.188
+        model: MSN2700 - CS2FO
+        serial: MT1234X56789
+        base_mac: 24:8a:07:12:34:56
+        syseeprom_info:
+            "0x21": "MSN2700"
+            "0x22": "MSN2700-CS2FO"
+            "0x23": "MT1234X56789"
+            "0x24": "24:8a:07:12:34:56"
+            "0x25": "12/07/2016"
+            "0x26": "0"
+            "0x28": "x86_64-mlnx_x86-r0"
+            "0x29": "2016.11-5.1.0008-9600"
+            "0x2A": "128"
+            "0x2B": "Mellanox"
+            "0xFE": "0xFBA1E964"
+    """
+    if 'syseeprom_info' in dut_vars:
```

Comment:
I think we should expect the `syseeprom_info` field to be present in the inventory file to prevent skipping this test and potentially missing an issue. Maybe we should assert here and fail the test if not present?

Reply:
I was trying to keep it backward compatible. With your proposed change, it will enforce everybody to add syseeprom_info for all their devices into the inventory file. Is that what we want to do.

Reply:
Yes, that's what I'm inferring. In order to have a full, comprehensive test, `syseeprom_info` data should be added to the inventory file. I think the test should fail if there's nothing to test against. We are currently in the process of improving test coverage, so I think this assumption in fine moving forward. Thank you for considering backward compatibility here, but when it comes to testing, we need to set some ground rules.